### PR TITLE
Updated fitting docstrings

### PR DESCRIPTION
**Pull request by karllark**
Updated the docstrings for the fitting subdirectory.  Used the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html) suggestions.

Also ran `black` on all the fitting code to remove PEP8 issues.

Docstings for all the routines now pass the API auto formatting so also updated the fitting API doc section.

Partially addresses (#80).

### CODE REVIEW

File: beast/fitting/pdf2d.py [hunk 1]
```diff
@@ -19,29 +19,26 @@ def __init__(
...
-            number of bins to use for each dimension of the 2D PDF
-
-        logspacing_p1, logspacing_p2: bool
-            whether to use logarithmic spacing for the bins of each dimension
-
-        minval_p1, maxval_p1, minval_p2, maxval_p2: float (optional)
+        gridvals_p1, gridvals_p2 : ndarray
+            1D `float` array with the values of the quantity for all the grid points
+        nbins_p1, nbins_p2 : int
+            number of bins to use for the 1D pdf
```
Comment (galaxyumi) [inline beast/fitting/pdf2d.py:30]: not 2D?

File: beast/fitting/pdf2d.py [hunk 2]
```diff
@@ -97,7 +102,7 @@ def __init__(
         for i in range(self.nbins_p1):
             for j in range(self.nbins_p2):
                 # find the indicies for the current bin
-                cur_bin_indxs, = np.where(
+                (cur_bin_indxs,) = np.where(
```
Comment (galaxyumi) [inline beast/fitting/pdf2d.py:105]: what does parenthesis do here?

File: beast/fitting/pdf2d.py [hunk 3]
```diff
@@ -114,15 +119,30 @@ def __init__(
...
     def gen2d(self, gindxs, weights):
+        """
+        Compute the 2D posterior PDFs based on the nD probabilities
+
+        Parameters
+        ----------
+        gindxs : ndarray
+            1D `int` array with the indxs of the weights in the full model grid
+        weights : ndarray
+            1D `float` array with the fit proabilities (likelihood*prior)
```
Comment (galaxyumi) [inline beast/fitting/pdf2d.py:131]: typo: probabilities

File: beast/fitting/trim_grid.py [hunk 4]
```diff
@@ -29,33 +29,25 @@ def trim_models(
...
         observation catalog
-
     sed_outname: str
         name for output sed file
-
     noisemodel_outname: str
         name for output noisemodel file
-
-    sigma_fac: float
+    sigma_fac: float, optional
```
Comment (galaxyumi) [inline beast/fitting/trim_grid.py:40]: want a space between parameter name and : to be consistent with your updates above?

File: beast/fitting/pdf2d.py (hunk 1 above)
Comment (karllark) [reply beast/fitting/pdf2d.py:30]: Nope.  The model SED grid is just a list of models and parameters.  Hence, when the grid values are passed, they are 1D arrays.  They are in order, so the correlations are preserved.

File: beast/fitting/pdf2d.py (hunk 2 above)
Comment (karllark) [reply beast/fitting/pdf2d.py:105]: The `black` autoformatting code adds them.  Some PEP statement about this somewhere I would assume.  Makes it explicit and easier to read that is a tuple decomposition I would assume.

File: beast/fitting/pdf2d.py (hunk 3 above)
Comment (karllark) [reply beast/fitting/pdf2d.py:131]: Fixed.

File: beast/fitting/trim_grid.py (hunk 4 above)
Comment (karllark) [reply beast/fitting/trim_grid.py:40]: Yep.

File: beast/fitting/pdf2d.py (hunk 2 above)
Comment (galaxyumi) [reply beast/fitting/pdf2d.py:105]: Aha!

**Pull request merged**